test/lsp/multithreaded_protocol_test_corpus.cc

```diff
@@ -618,8 +615,8 @@ TEST_CASE_FIXTURE(MultithreadedProtocolTest, "CanceledRequestsDontReportLatencyM
     sendAsync(LSPMessage(make_unique<NotificationMessage>("2.0", LSPMethod::PAUSE, nullopt)));
     // Send a hover to request the documentation string.
     sendAsync(*hover("foo.rb", 2, 6));
-    // Pause so that all latency timer for the hover is >1ms and would normally get reported.
-    this_thread::sleep_for(chrono::milliseconds(2));
+    // Ensure all latency timers are reported.
+    Timer::disableDroppingShortTimersForTests();
```

Comment:
We're calling this in four places but never cleaning up after ourselves. Are we introducing flakiness into these tests?

I think we should either set this once at the start of the test and have it apply to everything, or have something that automatically cleans up the state when a variable goes out of scope.

Also I'm not super familiar with this test, but it says `multithreaded` in the name. This threshold is not stored in a threadlocal and is not guarded by synchronization code, but we're mutating it multiple times. Maybe that's fine, maybe it's a problem—jvilk would know more.

Reply:
> Also I'm not super familiar with this test, but it says multithreaded in the name. This threshold is not stored in a threadlocal and is not guarded by synchronization code, but we're mutating it multiple times. Maybe that's fine, maybe it's a problem—jvilk would know more.

Good point; I should've caught this. We should be calling this once at the start of the test suites that involve timers.